ethsigner/signer/multiplatform/src/main/java/tech/pegasys/ethsigner/signer/multiplatform/SigningMetadataTomlConfigLoader.java

```diff
@@ -73,28 +72,32 @@
     }
   }
 
-  Optional<FileBasedSigningMetadataFile> getMetadataInfo(final Path file) {
+  private Optional<FileBasedSigningMetadataFile> getMetadataInfo(final Path file) {
     try {
       final TomlParseResult result =
           TomlConfigFileParser.loadConfigurationFromFile(file.toAbsolutePath().toString());
       final String type = result.getTable("signing").getString("type");
-      if (FILE_BASED_SIGNER.equals(type)) {
-        final String keyFilename = result.getTable("signing").getString("key-file");
-        final Path keyPath = new File(keyFilename).toPath();
-        final String passwordFilename = result.getTable("signing").getString("password-file");
-        final Path passwordPath = new File(passwordFilename).toPath();
-        return Optional.of(
-            new FileBasedSigningMetadataFile(file.getFileName().toString(), keyPath, passwordPath));
+      if (SignerType.FILE_BASED_SIGNER.getType().equals(type)) {
+        return getFileBasedSigningMetadataFromToml(file.getFileName().toString(), result);
       } else {
         LOG.error("Unknown signing type in metadata: " + type);
         return Optional.empty();
       }
-    } catch (Exception e) {
+    } catch (final Exception e) {
       LOG.error("Could not load TOML file", e);
       return Optional.empty();
     }
   }
 
+  private Optional<FileBasedSigningMetadataFile> getFileBasedSigningMetadataFromToml(
+      String filename, TomlParseResult result) {
+    final String keyFilename = result.getTable("signing").getString("key-file");
+    final Path keyPath = new File(keyFilename).toPath();
+    final String passwordFilename = result.getTable("signing").getString("password-file");
+    final Path passwordPath = new File(passwordFilename).toPath();
+    return Optional.of(new FileBasedSigningMetadataFile(filename, keyPath, passwordPath));
```

Comment:
can we remove the "Optional" - i.e. this function will always produce a FIleBasedSigningMetadataFile?

Reply:
do you mean return null if there's an error? I thought Optional was better than that